nav2_planner/src/planner_server.cpp

```diff
@@ -282,65 +282,32 @@ void PlannerServer::getPreemptedGoalIfRequested(
 
 template<typename T>
 bool PlannerServer::getStartPose(
-  std::unique_ptr<nav2_util::SimpleActionServer<T>> & action_server,
```

Comment:
Why remove the action server? This is strictly necessary to terminate in the places you removed them from.

Reply:
Yeah, the action server still gets terminated. If `getStartPose` returns false, we throw a plannerTFError which gets caught and handled.

Reply:
Ah makes sense! What about removing `validatePath`? That's a useful abstraction and could have a similar thing (if returns false, then throws and catches to terminate)

Reply:
Well that func only checks to see if the path was non empty. So that got changed to look like

` 
if (result->path.poses.empty()) {
     throw nav2_core::PlannerException(goal->planner_id + "generated a empty path");
}`

Reply:
But that was abstracting out logic from both the plan to pose and plan through poses APIs, and likely is to be expanded over time for additional checks. Please readd the function even if for now it just logs / checks for the path being empty to reduce code duplication

Reply:
added it back in :)